pkg/registry/utils.go

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"golang.org/x/crypto/bcrypt"
+	"golang.org/x/sync/errgroup"
+	"net"
+)
+
+func GenerateHTTPBasicAuth(username, password string) (string, error) {
```

Comment:
maybe put at pkg/utils.

Reply:
It seems that this function `GenerateHTTPBasicAuth` will only be used in the registry pkg, so I did not put it in the public utils.